### PR TITLE
Add SettingsMixin and default values population from config files

### PR DESCRIPTION
**Pull request by z4y4ts**

### CODE REVIEW

File: video_xblock/mixins.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+@XBlock.wants('settings')
+class SettingsMixin(XBlock):
+    """
+    SettingsMixin provides access to XBlock settings service.
+
+    Provides convenient access to XBlock's settings set in edx-platform config files.
+
+    Sample default settings in /edx/app/edxapp/cms.env.json & /edx/app/edxapp/lms.env.json files:
+    "XBLOCK_SETTINGS": {
```
Comment (xahgmah) [inline video_xblock/mixins.py]: Should be described on README.md too

File: video_xblock/mixins.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+                }
+        """
+        s_service = self.runtime.service(self, 'settings')
+        if s_service:
+            # At the moment SettingsService is not available in the context
+            # of Studio Edit modal. See https://github.com/edx/edx-platform/pull/14648
+            return s_service.get_settings_bucket(self)
+        else:
+            from django.conf import settings
+            return settings.XBLOCK_SETTINGS[self.block_settings_key]
```
Comment (xahgmah) [inline video_xblock/mixins.py]: Shouldn't it be `settings.XBLOCK_SETTINGS.get(self.block_settings_key, {})` ?

Comment (z4y4ts) [inline video_xblock/mixins.py]: Yeah, missed this. Good catch!

File: video_xblock/mixins.py (hunk 1 above)
Comment (z4y4ts) [inline video_xblock/mixins.py]: Fixed

File: video_xblock/mixins.py (hunk 2 above)
Comment (z4y4ts) [inline video_xblock/mixins.py]: Changed

**Pull request merged**